diagnosis/collectlogs.sh

```diff
@@ -9,46 +9,57 @@ CURRENTUSER=`whoami`
+log_level -i "[$HOSTNAME] Starting log collection" | tee -a $TRACEFILENAME
 
-echo "[$(date +%Y%m%d%H%M%S)][INFO][$HOSTNAME] Looking for syslog file" | tee -a $TRACEFILENAME
+log_level -i "[$HOSTNAME] Looking for syslog file" | tee -a $TRACEFILENAME
 try_copy_file /var/log/syslog $LOGDIRECTORY/
 
-echo "[$(date +%Y%m%d%H%M%S)][INFO][$HOSTNAME] Dumping Microsoft Azure Linux Agent (waagent) directory tree" | tee -a $TRACEFILENAME
+log_level -i "[$HOSTNAME] Dumping Microsoft Azure Linux Agent (waagent) directory tree" | tee -a $TRACEFILENAME
 try_print_directory_tree /var/lib/waagent $LOGDIRECTORY/waagent.tree
 
-echo "[$(date +%Y%m%d%H%M%S)][INFO][$HOSTNAME] Looking for Microsoft Azure Linux Agent (waagent) log file" | tee -a $TRACEFILENAME
+log_level -i "[$HOSTNAME] Looking for Microsoft Azure Linux Agent (waagent) log file" | tee -a $TRACEFILENAME
 try_copy_file /var/log/waagent.log $LOGDIRECTORY/
 
-echo "[$(date +%Y%m%d%H%M%S)][INFO][$HOSTNAME] Looking for cloud-init log files" | tee -a $TRACEFILENAME
+log_level -i "[$HOSTNAME] Looking for cloud-init log files" | tee -a $TRACEFILENAME
 try_copy_file /var/log/cloud-init.log $LOGDIRECTORY/
 try_copy_file /var/log/cloud-init-output.log $LOGDIRECTORY/
 
-echo "[$(date +%Y%m%d%H%M%S)][INFO][$HOSTNAME] Looking for CSE directory" | tee -a $TRACEFILENAME
+log_level -i "[$HOSTNAME] Looking for CSE directory" | tee -a $TRACEFILENAME
 mkdir -p $LOGDIRECTORY/cse/
 try_copy_directory_content /var/log/azure/ $LOGDIRECTORY/cse
 try_copy_file /opt/m $LOGDIRECTORY/cse/
 
-echo "[$(date +%Y%m%d%H%M%S)][INFO][$HOSTNAME] Looking for apt logs" | tee -a $TRACEFILENAME
+log_level -i "[$HOSTNAME] Looking for apt logs" | tee -a $TRACEFILENAME
 mkdir -p $LOGDIRECTORY/apt/
 try_copy_directory_content /var/log/apt/ $LOGDIRECTORY/apt
 
-echo "[$(date +%Y%m%d%H%M%S)][INFO][$HOSTNAME] Dumping running container list" | tee -a $TRACEFILENAME
+log_level -i "[$HOSTNAME] Dumping running container list" | tee -a $TRACEFILENAME
 sudo docker ps &> $LOGDIRECTORY/containers.list
 
-echo "[$(date +%Y%m%d%H%M%S)][INFO][$HOSTNAME] Looking for containers logs" | tee -a $TRACEFILENAME
+log_level -i "[$HOSTNAME] Looking for containers logs" | tee -a $TRACEFILENAME
 mkdir -p $LOGDIRECTORY/containers/manifests
 
 test $# -gt 0 && NAMESPACES=$@
-test -z "${NAMESPACES}" && echo "[$(date +%Y%m%d%H%M%S)][INFO][$HOSTNAME] Collection logs from all namespaces"
-test -n "${NAMESPACES}" && echo "[$(date +%Y%m%d%H%M%S)][INFO][$HOSTNAME] Collection logs from containers in these namespaces: $NAMESPACES."
+test -z "${NAMESPACES}" && log_level -i "[$HOSTNAME] Collection logs from all namespaces"
+test -n "${NAMESPACES}" && log_level -i "[$HOSTNAME] Collection logs from containers in these namespaces: $NAMESPACES."
+
+#collecting pod information 
+PODDIR=$(ls /var/log/pods)
+log_level -i "[$HOSTNAME] Collecting Pod Logs" | tee -a $TRACEFILENAME
+if [[ ! -z $PODDIR ]];
```

Comment:
We need to have way to collect data only from kube-system or passed namespaces.

Reply:
is this potentially collecting the same log twice?